activerecord/test/cases/finder_test.rb

```diff
@@ -987,8 +1001,11 @@ def test_find_last_with_offset
     devs = Developer.order('id')
 
     assert_equal devs[2], Developer.offset(2).first
-    assert_equal devs[-3], Developer.offset(2).last
-    assert_equal devs[-3], Developer.offset(2).last
+    assert_equal devs[-1], Developer.offset(2).last
+    assert_deprecated { Developer.offset(2).last }
+    # assert_raise(ActiveRecord::IrreversibleOrderError) { Developer.offset(2).last } # Rails >= 5.1
+    assert_deprecated { Developer.offset(1).reverse_order }
```

Comment:
Great point. That can be organized much better. I’ve made the change in 7a7da3f.